balancer-js/src/modules/relayer/actions.ts

```diff
@@ -0,0 +1,204 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { MaxInt256 } from '@ethersproject/constants';
+import { defaultAbiCoder } from '@ethersproject/abi';
+import { BatchSwapStep, SwapType } from '../swaps/types';
+import { Relayer } from './relayer.module';
+import { OutputReference } from './types';
+
+export const gaugeWithdrawal = Relayer.encodeGaugeWithdraw;
+export const gaugeDeposit = Relayer.encodeGaugeDeposit;
+export const peekChainedReferenceValue =
+  Relayer.encodePeekChainedReferenceValue;
+
+/**
+ * Encodes exitPool callData.
+ * Exit pool to underlying tokens and assigns output references to the exit amounts.
+ *
+ * @param poolId Pool ID.
+ * @param assets Ordered pool tokens.
+ * @param singleTokenExit When
+ * @param outputReferences reference to exit amounts for the next transaction
+ * @param amount Amount of BPT to exit with as a number with 18 digits of precision passed as a string.
+ * @param sender Sender address.
+ * @param recipient Recipient address.
+ * @param isComposable Whether the poolType is ComposableStable or not.
+ * @param toInternalBalance Use internal balance or not.
+ * @returns Encoded exitPool call.
+ */
+export const exit = (
+  poolId: string,
+  assets: string[],
+  exitTokenIndex = -1,
+  outputReferences: OutputReference[],
+  amount: string,
+  sender: string,
+  recipient: string,
+  isComposable = false,
+  toInternalBalance = true
+): string => {
+  let userData: string;
+
+  // Exit pool proportionally or to a singleToken
+  if (exitTokenIndex > -1) {
+    userData = defaultAbiCoder.encode(
+      ['uint256', 'uint256', 'uint256'],
+      [0, amount, exitTokenIndex]
+    );
+  } else {
+    const exitType = isComposable ? 2 : 1;
+    userData = defaultAbiCoder.encode(
```

Comment:
How about we use named encodings instead of anonymous ones? You should be able to reuse `StablePoolEncoder` if I'm not mistaken.

Reply:
> I had to go through it twice to understand it all 😅 Looks good 👍

ok, we aim for just one read-through  ;)

Reply:
@brunoguerios - replaced generic encoders with our pool classes.